tests/unit/completion/test_completionwidget.py

```diff
@@ -0,0 +1,152 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for the CompletionView Object."""
+
+import unittest.mock
+
+import pytest
+from PyQt5.QtGui import QStandardItem, QColor
+
+from qutebrowser.completion import completionwidget
+from qutebrowser.completion.models import base, sortfilter
+from qutebrowser.utils import objreg
+
+
+@pytest.yield_fixture
+def completionview(qtbot, status_command_stub, config_stub, win_registry,
+                   monkeypatch):
+    """Create the CompletionView used for testing."""
+    config_stub.data = {
+        'completion': {
+            'show': True,
+            'auto-open': True,
+            'scrollbar-width': 12,
+            'scrollbar-padding': 2,
+            'shrink': False,
+        },
+        'colors': {
+            'completion.fg': QColor(),
+            'completion.bg': QColor(),
+            'completion.alternate-bg': QColor(),
+            'completion.category.fg': QColor(),
+            'completion.category.bg': QColor(),
+            'completion.category.border.top': QColor(),
+            'completion.category.border.bottom': QColor(),
+            'completion.item.selected.fg': QColor(),
+            'completion.item.selected.bg': QColor(),
+            'completion.item.selected.border.top': QColor(),
+            'completion.item.selected.border.bottom': QColor(),
+            'completion.match.fg': QColor(),
+            'completion.scrollbar.fg': QColor(),
+            'completion.scrollbar.bg': QColor(),
+        },
+        'fonts': {
+            'completion': 'Comic Sans Monospace'
+        }
+    }
+    # mock the Completer that the widget creates in its constructor
+    monkeypatch.setattr('qutebrowser.completion.completer.Completer',
+        lambda *args: unittest.mock.Mock())
+    view = completionwidget.CompletionView(win_id=0)
+    qtbot.addWidget(view)
+    yield view
+    # the constructor registers both 'completion' (itself) and 'completer'
+    objreg.delete('completion', scope='window', window=0)
+    objreg.delete('completer', scope='window', window=0)
+
+
+def test_set_model(completionview):
+    """Ensure set_model actually sets the model and expands all categories."""
+    model = base.BaseCompletionModel()
+    filtermodel = sortfilter.CompletionFilterModel(model)
+    for i in range(3):
+        model.appendRow(QStandardItem(str(i)))
+    completionview.set_model(filtermodel)
+    assert completionview.model() == filtermodel
```

Comment:
I guess `==` does the same here (as it's probably not implemented for models), but since you're actually comparing object identies here, `is` would be better.